x-pack/plugins/security/server/authentication/authenticator.ts

```diff
@@ -325,6 +327,14 @@ export class Authenticator {
       }
     }
 
+    // If there was an attempt to log in, but it couldn't be handled, we should invalidate existing
```

Comment:
__note:__ noticed this confusing thing while testing: 

Log in with native user, and then try to log in via SAML IdP initiated login using realm that's not configured. In this case SAML authentication provider will return `notHandled` giving opportunity to another SAML provider in case SAMLResponse is targeting another SAML realm. This login flow will eventually complete with `notHandled` result and we'll redirect you to the login page, but because of the existing and valid session login page will immediately redirect you to a home page. So you may have impression that you successfully logged in with SAML :facepalm: